commit a0642b8236c9faa6fe6498d8f3a7f37a9d7bc9bd
Author: Seth Forshee <seth.forshee@canonical.com>
Date:   Wed Oct 7 14:49:47 2015 -0500

    UBUNTU: SAUCE: (namespace) block_dev: Check permissions towards block device inode when mounting
    
    Unprivileged users should not be able to mount block devices when
    they lack sufficient privileges towards the block device inode.
    Update blkdev_get_by_path() to validate that the user has the
    required access to the inode at the specified path. The check
    will be skipped for CAP_SYS_ADMIN, so privileged mounts will
    continue working as before.
    
    Acked-by: Serge Hallyn <serge.hallyn@canonical.com>
    Signed-off-by: Seth Forshee <seth.forshee@canonical.com>
    Signed-off-by: Tim Gardner <tim.gardner@canonical.com>

diff --git a/fs/block_dev.c b/fs/block_dev.c
index ebf0453..0bfe3e2 100644
--- a/fs/block_dev.c
+++ b/fs/block_dev.c
@@ -1496,9 +1496,14 @@ struct block_device *blkdev_get_by_path(const char *path, fmode_t mode,
 					void *holder)
 {
 	struct block_device *bdev;
+	int perm = 0;
 	int err;
 
-	bdev = lookup_bdev(path, 0);
+	if (mode & FMODE_READ)
+		perm |= MAY_READ;
+	if (mode & FMODE_WRITE)
+		perm |= MAY_WRITE;
+	bdev = lookup_bdev(path, perm);
 	if (IS_ERR(bdev))
 		return bdev;
 
